src/ol_infrastructure/applications/mitxpro/__main__.py

```diff
@@ -0,0 +1,216 @@
+    name=f"mitxpro-edxapp-access-{env_suffix}",
+    description="Access control to mitxpro_edxapp",
+    ingress=[
+        ec2.SecurityGroupIngressArgs(
+            cidr_blocks=[mitxpro_vpc["cidr"]],
+            ipv6_cidr_blocks=[mitxpro_vpc["cidr_v6"]],
+            protocol="tcp",
+            from_port=22,
+            to_port=22,
+            description="mitxpro_vpc ssh access",
+        ),
+        ec2.SecurityGroupIngressArgs(
+            cidr_blocks=[operations_vpc["cidr"]],
+            ipv6_cidr_blocks=[operations_vpc["cidr_v6"]],
+            protocol="tcp",
+            from_port=22,
+            to_port=22,
+            description="operations_vpc ssh access",
+        ),
+        ec2.SecurityGroupIngressArgs(
+            cidr_blocks=["0.0.0.0/0"],
+            ipv6_cidr_blocks=["::/0"],
+            protocol="tcp",
+            from_port=80,
+            to_port=80,
+            description="HTTP access",
+        ),
+        ec2.SecurityGroupIngressArgs(
+            cidr_blocks=["0.0.0.0/0"],
+            ipv6_cidr_blocks=["::/0"],
+            protocol="tcp",
+            from_port=443,
+            to_port=443,
+            description="HTTPS access",
+        ),
+        ec2.SecurityGroupIngressArgs(
+            cidr_blocks=[mitxpro_vpc["cidr"]],
+            ipv6_cidr_blocks=[mitxpro_vpc["cidr_v6"]],
+            protocol="tcp",
+            from_port=18040,
+            to_port=18040,
+            description="Xqueue access",
+        ),
+    ],
+    tags=aws_config.tags,
+    vpc_id=mitxpro_vpc["id"],
+)
+
+mitxpro_edx_worker_security_group = ec2.SecurityGroup(
+    f"mitxpro-edx_worker-access-{env_suffix}",
+    name=f"mitxpro-edx_worker-access-{env_suffix}",
+    description="Access control to mitxpro_edx_worker",
+    ingress=[
+        ec2.SecurityGroupIngressArgs(
+            security_groups=[mitxpro_edxapp_security_group.id],
+            protocol="tcp",
+            from_port=18040,  # noqa: WPS432
+            to_port=18040,  # noqa: WPS432
+            description="Xqueue access",
+        )
+    ],
+    tags=aws_config.tags,
+    vpc_id=mitxpro_vpc["id"],
+)
+
+mitxpro_edxapp_db_security_group = ec2.SecurityGroup(
+    f"mitxpro-edxapp-db-access-{env_suffix}",
+    name=f"mitxpro-edxapp-db-access-{env_suffix}",
+    description="Access from the mitxpro VPC to the mitxpro edxapp database",
+    ingress=[
+        ec2.SecurityGroupIngressArgs(
+            security_groups=[
+                mitxpro_edxapp_security_group.id,
+                mitxpro_edx_worker_security_group.id,
+                consul_server_security_group.id,
+            ],
+            protocol="tcp",
+            from_port=3306,  # noqa: WPS432
+            to_port=3306,  # noqa: WPS432
+        )
+    ],
+    tags=aws_config.tags,
+    vpc_id=mitxpro_vpc["id"],
+)
+
+mitxpro_edxapp_db_config = OLMariaDBConfig(
+    instance_name=f"ol-mitxpro-edxapp-db-{env_suffix}",
+    password=get_config("mitxpro_edxapp:db_password"),
```

Comment:
The `get_config` method will happily allow you to not specify the corresponding stack value. We should get in the habit of using the `Config` class object and the associated methods such as `require_secret`, `require_object`, etc.

So for this it would by:
```python
from pulumi import Config

xpro_config = Config('mitxpro_edxapp')
...
password=mitxpro_config.require_secret('db_password')
...
db_purpose=mitxpro_config.require('db_purpose')
```